### PR TITLE
Move swarming docs

### PR DESCRIPTION
**Pull request by wdm-slc**
Also see https://github.com/SkylineCommunications/internal-docs/pull/1686

### CODE REVIEW

File: user-guide/Advanced_Functionality/Swarming/EnableSwarming.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+- Enhanced services in the cluster (if any) use only [compatible connectors](xref:SwarmingPrepare).
+- Scripts (Automation/GQI) and QActions in connectors have all been [made compatible](xref:SwarmingPrepare), i.e. they do not use obsolete or incompatible SLNet calls/properties that handle Alarm IDs as "DmaID/AlarmID".
+- For the user account that will be used to enable Swarming, the [Admin Tools](xref:DataMiner_user_permissions#modules--system-configuration--tools--admin-tools) user permission is enabled.
+
+## Running a prerequisites check
+
+To verify whether your system meets these prerequisites before you enable Swarming, you can send a *SwarmingPrerequisitesCheckRequest* message to SLNet using the [SLNetClientTest tool](xref:SLNetClientTest_tool). The prerequisite check is also executed when you enable Swarming, ensuring that all prerequisites are met before Swarming is enabled.
+
+> [!NOTE]
+> The prerequisites check can take some time, as checking the usage of legacy alarm IDs in scripts can take a while.
```
Comment (RobbeDGH) [inline user-guide/Advanced_Functionality/Swarming/EnableSwarming.md:25]: should we give an indication?
eg "multiple minutes" 

"Some time" is a bit ambiguous and 20s in the context of slnet msg is also already "some time"...

File: user-guide/Advanced_Functionality/Swarming/EnableSwarming.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+To run a prerequisites check using SLNetClientTest tool:
+
+1. [Connect to the DMA using the SLNetClientTest tool](xref:Connecting_to_a_DMA_with_the_SLNetClientTest_tool).
+
+   > [!CAUTION]
+   > Always be extremely careful when using this tool, as it can have far-reaching consequences on the functionality of your DataMiner System.
+
+1. Go to the *Build Message* tab of the main window of the SLNetCLientTest tool.
+
+1. In the *Message Type* drop-down list, select the message *Skyline.DataMiner.Net.Swarming.SwarmingPrerequisitesCheckRequest*.
```
Comment (RobbeDGH) [inline user-guide/Advanced_Functionality/Swarming/EnableSwarming.md:36]: should we guide the user in skipping the alarmid check if they want to check the other things first or later on repeat?

File: user-guide/Advanced_Functionality/Swarming/EnableSwarming.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+1. In the *Message Type* drop-down list, select the message *Skyline.DataMiner.Net.Swarming.EnableSwarmingRequest*.
+
+   This message will need to be sent with its default settings, so do not make any changes to the settings for the message.
+
+1. Click *Send Message*.
+
+   The prerequisites will be checked for all Agents in the system, and when they are met, Swarming will enabled and all Agents in the DMS will be **restarted**.
+
+   The prerequisites check can take some time, as checking the usage of legacy alarm IDs in scripts can take a while.
```
Comment (RobbeDGH) [inline user-guide/Advanced_Functionality/Swarming/EnableSwarming.md:72]: idem

File: user-guide/Advanced_Functionality/Swarming/EnableSwarming.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+## Verifying that the feature is active
+
+Once all Agents have been restarted, you can verify whether the Swarming feature has been activated by checking if the following line is present in the *SLSwarming.txt* log file:
+
+`SwarmingManager::SwarmingManager|INF|-1|Swarming enabled`
+
+You can find this log file in DataMiner Cube via *System Center* > *Logging* > *dataminer* > *Swarming*. The log file also shows details on the migration of elements that happened during DMA startup.
+
+Once Swarming is enabled, you can go to *System Center* > *Agents* > *Status* > *Swarm* to swarm elements for which this feature is supported.
```
Comment (RobbeDGH) [inline user-guide/Advanced_Functionality/Swarming/EnableSwarming.md:87]: Note: only in clusters. Single agents do not show this button, even if swarming is enabled

File: user-guide/Advanced_Functionality/Swarming/PrepareSwarming.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+- **Alarm Tree ID**: Reference which can be used to uniquely refer to an alarm tree without having to refer to individual alarm events. For example, when adding a comment to an alarm tree, one can send the alarm tree ID to the server to apply the update to the most recent state of that alarm tree. An ID looks like `111/222/333` where `111/222` is a reference to the element involved. The Alarm Tree ID is the preferred way to reference alarm trees.
+- **Root Alarm ID**: Legacy number (e.g. `333`) indicating a specific alarm tree. Not guaranteed to be unique across elements or agents.
+- **Alarm ID**: Number referencing an alarm event within an alarm tree (e.g. `444`). Not to be used on its own. Can be used in combination with an alarm tree ID in cases where an individual alarm event needs to be referenced. Try to avoid these ([more info](#alarmid-references)). For root alarm events, the alarm ID is identical to the root alarm ID.
+- **Hosting Agent ID**: DataMiner ID of the agent on which the element owning the alarm event/tree is currently hosted.
```
Comment (RobbeDGH) [inline user-guide/Advanced_Functionality/Swarming/PrepareSwarming.md:20]: not completely/always true right?

Its the agent that was hosting the element when the alarm was created?

File: user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+When Swarming is enabled, the element config XML files (eg `Element.xml`, `ElementData.xml`, `Description.xml`, etc.) are moved from disk to the database. There is currently no procedure to move them back to disk.
+
+There are some things you can do partially recover some data.
+
+## Partial Rollback from backup
+
+When DataMiner is moving the XML files from disk to database, it will take a backup first. This migration happens during first startup after enabling the Swarming feature.
+
+The backup gets initially stored in `Recycle Bin`, but could be manually copied over to a different locationto have long term access.
```
Comment (RobbeDGH) [inline user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md:18]: missing space in locationto

File: user-guide/Advanced_Functionality/Swarming/Swarming.md [hunk 7]
```diff
@@ -4,25 +4,80 @@ uid: Swarming
...
+
+|                                                  | 10.5.1+&nbsp; systems without Swarming | 10.5.1+ systems with Swarming  |
+|--------------------------------------------------|----------------------------------------|--------------------------------|
+| Element configuration                            | On disk (element.xml)                  | In database                    |
+| Alarm IDs                                        | Per DataMiner Agent                    | Per element                    |
+| Database per DMA                                 | Supported                              | Not supported                  |
+| Scripts & QActions using legacy alarm references | Supported                              | Not supported                  |
+| Legacy Reports & Dashboards                      | Supported                              | Not supported                  |
+| Failover                                         | Supported                              | Not supported                  |
+| Legacy Correlation                               | Not supported                          | Not supported                  |
```
Comment (RobbeDGH) [inline user-guide/Advanced_Functionality/Swarming/Swarming.md:60]: should we still mention this as this is removed entirely in 10.5.1?

File: user-guide/Advanced_Functionality/Swarming/Swarming.md [hunk 8]
```diff
@@ -4,25 +4,80 @@ uid: Swarming
...
+| Spectrum elements                                | Supported                              | Supported but not swarmable*   |
+| Redundancy group elements                        | Supported                              | Supported but not swarmable*   |
+| Elements exporting DVEs of functions             | Supported                              | Supported but not swarmable*   |
+| EPM elements                                     | Supported                              | Supported but not swarmable*   |
+| Elements polling localhost                       | Supported                              | Supported but not swarmable    |
+| Elements with element connections                | Supported                              | Supported but not swarmable*   |
+| Smart-serial elements in server mode             | Supported                              | Supported but not swarmable    |
+| Elements receiving SNMP traps                    | Supported                              | Supported but not swarmable    |
+
+(*) In initial release.
```
Comment (RobbeDGH) [inline user-guide/Advanced_Functionality/Swarming/Swarming.md:73]: perhaps we should reword this to 
(*) To be added later

Because now you might expect it to be added in 10.5.2 or something?

File: user-guide/Advanced_Functionality/Swarming/SwarmingElements.md [hunk 9]
```diff
@@ -4,17 +4,150 @@ uid: SwarmingElements
 
 # Swarming elements
 
-With DataMiner Swarming, you can swarm basic elements from one DataMiner Agent in a cluster to another. At present, the *Swarming* [soft-launch option](xref:SoftLaunchOptions) must be enabled for this.
+With DataMiner Swarming, you can swarm basic elements from one DataMiner Agent in a cluster to another.
```
Comment (RobbeDGH) [inline user-guide/Advanced_Functionality/Swarming/SwarmingElements.md:7]: suggestion: 
from one DataMiner Agent to another within a cluster

File: user-guide/Advanced_Functionality/Swarming/SwarmingElements.md [hunk 10]
```diff
@@ -4,17 +4,150 @@ uid: SwarmingElements
...
+During this transition, the ability to open element cards or change element configuration for the involved element will be temporarily suspended. Once the element migration is complete, it will become accessible again.
+
+> [!NOTE]
+> To be able to trigger swarming for an element, you need the *Swarming* user permission as well as config rights on the element.
+
+## Swarming an element from Cube
+
+From Cube, elements can be swarmed from *System Center > Agents > Status > Swarm...*
+
+## Swarming an element from Automation
```
Comment (RobbeDGH) [inline user-guide/Advanced_Functionality/Swarming/SwarmingElements.md:25]: add reference to catalog package with such a script?
https://catalog.dataminer.services/details/ffb0166d-9394-4f14-abd0-48e2175484a0

File: user-guide/Advanced_Functionality/Swarming/PrepareSwarming.md [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+- **Root alarm ID**: Legacy number (e.g. `333`) indicating a specific alarm tree. This is not guaranteed to be unique across elements or DMAs.
+- **Alarm ID**: Number referencing an alarm event within an alarm tree (e.g. `444`). This should not be used on its own, but it can be used in combination with an alarm tree ID in cases where an individual alarm event needs to be referenced. Try to avoid such alarm ID references ([more info](#alarmid-references)). For root alarm events, the alarm ID is identical to the root alarm ID.
+- **Hosting Agent ID**: The DataMiner ID of the DMA hosting the element that the alarm event/tree is linked to.
+
+In DataMiner versions prior to DataMiner 10.4, the combination `HostingAgentID/RootAlarmID` is typically used to refer to an alarm tree. When elements are [migrated between DMAs](xref:Migrating_elements_in_a_DataMiner_System) or swarmed from one DMA to another, this combination of fields is inadequate as it is not unique or fixed.
```
Comment (MariekeGO) [inline user-guide/Advanced_Functionality/Swarming/PrepareSwarming.md:26]: Which DataMiner versions introduce this change exactly? Is there an RN reference we can add here?

File: user-guide/Advanced_Functionality/Swarming/SwarmingElements.md (hunk 10 above)
Comment (RobbeDGH) [reply user-guide/Advanced_Functionality/Swarming/SwarmingElements.md]: or github repo?
https://github.com/SkylineCommunications/SLC-AS-SwarmElements

File: user-guide/Advanced_Functionality/Swarming/Swarming.md (hunk 7 above)
Comment (wdm-slc) [reply user-guide/Advanced_Functionality/Swarming/Swarming.md:60]: removed this one

File: user-guide/Advanced_Functionality/Swarming/Swarming.md (hunk 8 above)
Comment (wdm-slc) [reply user-guide/Advanced_Functionality/Swarming/Swarming.md:73]: Updated

File: user-guide/Advanced_Functionality/Swarming/PrepareSwarming.md (hunk 5 above)
Comment (wdm-slc) [reply user-guide/Advanced_Functionality/Swarming/PrepareSwarming.md]: updated

File: user-guide/Advanced_Functionality/Swarming/SwarmingElements.md (hunk 10 above)
Comment (wdm-slc) [reply user-guide/Advanced_Functionality/Swarming/SwarmingElements.md]: linked to both as a TIP

File: user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+1. Restart all DMAs.
+
+> [!IMPORTANT]
+> When you restore the backup, any modifications to existing elements or redundancy groups since the backup and any new elements or redundancy groups created since the backup will be lost. Any reference to these elements could become invalid.
+
+## Restoring elements using an export and import
+
+For small systems, an alternative is to use DELT to extract the latest version of the elements. You would then export the elements and reimport them once Swarming is disabled.
+
+1. [Export the elements](xref:Exporting_elements_services_etc_to_a_dmimport_file) you want to save to a .dmimport file.
```
Comment (MariekeGO) [inline user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md:42]: Is this something you should do before you enable Swarming, or can you do this when Swarming is already enabled? If it's the latter, perhaps we should rephrase the sentence below so it mentions "the element configuration" instead of "the element XML files"?

File: user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+
+   Do not include any data from the database in the export. Only the element XML files should be included.
+
+1. [Manually disable Swarming](#manually-disabling-swarming-in-the-config-files).
+
+1. [Import the elements](xref:Importing_elements_services_etc_from_a_dmimport_file) again.
+
+   The same element IDs must be retained in the import.
```
Comment (MariekeGO) [inline user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md:50]: Is there something the user has to do to make sure this happens?

Comment (wdm-slc) [reply user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md:50]: I'm not sure what this sentence means ("The same element IDs must be retained in the import."). 
I think it can be removed

File: user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md (hunk 12 above)
Comment (wdm-slc) [reply user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md:42]: It's something you can do while Swarming is enabled. Better to refer to element configuration indeed
A dmimport package taken while swarming is enabled can be restored on a system where swarming is not enabled

File: user-guide/Advanced_Functionality/Swarming/Swarming.md [hunk 14]
```diff
@@ -4,25 +4,80 @@ uid: Swarming
...
+| Legacy Reports & Dashboards                      | Supported                              | Not supported                  |
+| Failover                                         | Supported                              | Not supported                  |
+| Offload database                                 | Supported                              | Not supported*                 |
+| Contributing bookings                            | Supported                              | Not supported*                 |
+| SLA elements                                     | Supported                              | Supported but not swarmable*   |
+| Enhanced services                                | Supported                              | Supported** but not swarmable* |
+| Spectrum elements                                | Supported                              | Supported but not swarmable*   |
+| Redundancy group elements                        | Supported                              | Supported but not swarmable*   |
+| Elements exporting DVEs of functions             | Supported                              | Supported but not swarmable*   |
```
Comment (MariekeGO) [inline user-guide/Advanced_Functionality/Swarming/Swarming.md:66]: Should this be DVEs *or* functions? If yes, I'd rephrase this as follows to keep the terminology in line with the rest of the documentation:
```suggestion
| DVE child and parent elements and virtual function resources | Supported                              | Supported but not swarmable*   |
```

File: user-guide/Advanced_Functionality/Swarming/SwarmingElements.md [hunk 15]
```diff
@@ -4,17 +4,164 @@ uid: SwarmingElements
...
+> [!NOTE]
+>
+> - To be able to trigger swarming for an element, you need the *Swarming* user permission as well as config rights on the element.
+> - Swarming is currently only possible with regular elements. See [Limitations](xref:Swarming#limitations).
+
+## Swarming elements in DataMiner Cube
+
+To swarm elements in DataMiner Cube:
+
+1. Go to *System Center* > *Agents* > *Status* and click the *Swarm* button in the lower right corner.
```
Comment (MariekeGO) [inline user-guide/Advanced_Functionality/Swarming/SwarmingElements.md:27]: Does this replace the element migration functionality as described here: [https://docs.dataminer.services/user-guide/Advanced_Functionality/DataMiner_Systems/Migrating_elements_in_a_DataMiner_System.html](https://docs.dataminer.services/user-guide/Advanced_Functionality/DataMiner_Systems/Migrating_elements_in_a_DataMiner_System.html)? RN 38467 seems to suggest as much

Comment (MariekeGO) [reply user-guide/Advanced_Functionality/Swarming/SwarmingElements.md:27]: I've now added this info. However, one thing that still isn't clear and that users might wonder about is what advantages there might be to using swarming instead of migrating using DELT. Why should a user choose the one over the other?

File: user-guide/Advanced_Functionality/Swarming/Swarming.md (hunk 14 above)
Comment (MariekeGO) [reply user-guide/Advanced_Functionality/Swarming/Swarming.md:66]: Would this be OK? The original line does not mention that the exported DVEs/functions also can't be swarmed, but I assume this is also the case?
```suggestion
| DVE and virtual function child and parent elements            | Supported                              | Supported but not swarmable*   |
```

File: user-guide/Advanced_Functionality/Swarming/EnableSwarming.md (hunk 2 above)
Comment (MariekeGO) [reply user-guide/Advanced_Functionality/Swarming/EnableSwarming.md:36]: I've now added additional info in line 40 for this. Could you check if this is OK?

File: user-guide/Advanced_Functionality/Swarming/PrepareSwarming.md [hunk 16]
```diff
@@ -0,0 +1,199 @@
...
+| `GetCorrelationBaseAlarmDetailsMessage` | Use `AlarmEventID` instead of `AlarmID`. |
+| `GetAlarmLinksMessage` | Do not use. |
+
+## String references
+
+Often, tree references get stored in a string variable as `HostingAgentID + "/" + RootAlarmID`.
+
+- If possible, replace the `string` variable/field by an object of type `AlarmTreeID`.
+
+- Bring the `AlarmTreeID` as close to the location where an actual string reference is required.
```
Comment (MariekeGO) [inline user-guide/Advanced_Functionality/Swarming/PrepareSwarming.md:110]:
```suggestion
- Bring the `AlarmTreeID` as close as possible to the location where an actual string reference is required.
```

File: user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md [hunk 17]
```diff
@@ -0,0 +1,61 @@
+---
+uid: SwarmingRollback
+---
+
+# Partially rolling back Swarming
+
+> [!IMPORTANT]
+> There is currently no good way to disable Swarming and keep everything in the DMS intact. The info below only partially recovers some data. If you are using a test system and it does not matter if all data are lost, a clean reinstall is advised.
+
+When Swarming is enabled, the element config XML files (e.g. `Element.xml`, `ElementData.xml`, `Description.xml`, etc.) are moved from disk to the database. There is currently no procedure to move them back to disk.
```
Comment (MariekeGO) [inline user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md:10]: Is it just these 3 XML files that get moved to the database, or are there others as well? If it's just these 3, I'd remove the "etc.":
```suggestion
When Swarming is enabled, the element config XML files (e.g. `Element.xml`, `ElementData.xml`, and `Description.xml`) are moved from the disk to the database. There is currently no procedure to move them back to the disk.
```

File: user-guide/Advanced_Functionality/Swarming/EnableSwarming.md (hunk 2 above)
Comment (wdm-slc) [reply user-guide/Advanced_Functionality/Swarming/EnableSwarming.md:36]: ok for me

File: user-guide/Advanced_Functionality/Swarming/SwarmingElements.md (hunk 15 above)
Comment (wdm-slc) [reply user-guide/Advanced_Functionality/Swarming/SwarmingElements.md:27]: Migrating using DELT is the combination of a few steps:
- the element gets exported into a dmimport package (containing the element config)
- the element gets deleted on the original agent
- the DELT package gets copied over to the new agent
- the DELT dmimport package is imported on the new agent

Migrating using swarming:
- Element configuration remains in the shared database
- First agent unloads the element
- New agent loads the element

Migration using DELT is quite error-prone. For example, if the export and delete works but the copy/import does not, you would end up with a missing element. There's a lot less risk in the swarm action as it doesn't delete any configuration.

Does this help?

File: user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md (hunk 17 above)
Comment (wdm-slc) [reply user-guide/Advanced_Functionality/Swarming/RollbackSwarming.md:10]: There are also matrix discreets files (e.g. ports.xml). There's zero or more of these without fixed names (names as defined in the protocol -- https://docs.dataminer.services/develop/schemadoc/Protocol/Protocol.Params.Param.Measurement.Type-link.html

File: user-guide/Advanced_Functionality/Swarming/SwarmingElements.md (hunk 15 above)
Comment (MariekeGO) [reply user-guide/Advanced_Functionality/Swarming/SwarmingElements.md:27]: I've added a note with this info on [the migration page](https://github.com/SkylineCommunications/dataminer-docs/pull/3964/files#diff-fed342fec7ac30732731d004e4a91a6a89f06fede535eb96939e94e6d7e355b4) now. Does this look OK for you?
